### PR TITLE
[AIRFLOW-2559] Azure Fileshare hook

### PR DESCRIPTION
**Pull request by NielsZeilemaker**
Make sure you have checked _all_ steps below.

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2559
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:
An Azure Fileshare Hook

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [X] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by Fokko** [comment]
Please pay attention to the commit template: When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
Can you also extend this one: https://airflow.apache.org/integration.html#azure-microsoft-azure

Tnx

**Comment by NielsZeilemaker** [comment]
@fokko I've added the autoclass documentation

**Comment by kaxil** [comment]
@NielsZeilemaker You also need to add it over here: https://github.com/apache/incubator-airflow/blob/master/docs/code.rst

File: tests/contrib/hooks/test_azure_fileshare_hook.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        mock_instance = mock_service.return_value
+        hook = AzureFileShareHook(wasb_conn_id='wasb_test_sas_token')
+        hook.get_file('path', 'share', 'directory', 'file', max_connections=1)
+        mock_instance.get_file_to_path.assert_called_once_with(
+            'share', 'directory', 'file', 'path', max_connections=1
+        )
+
+    @mock.patch('airflow.contrib.hooks.azure_fileshare_hook.FileService',
+                autospec=True)
+    def get_file_to_stream(self, mock_service):
```
Comment (Fokko) [inline tests/contrib/hooks/test_azure_fileshare_hook.py]: I think this method should start with `test_`

**Review by Fokko — CHANGES_REQUESTED** [review]

**Comment by Fokko** [comment]
Small nits, apart from that LGTM

**Comment by NielsZeilemaker** [comment]
@Fokko I've incorporated the last two remarks. Should be ready to merge

**Comment by kaxil** [comment]
Thank You @NielsZeilemaker

**Pull request closed without merging**